cumulus/client/cirrus-executor/src/processor.rs

```diff
@@ -151,15 +153,25 @@ where
 		let extrinsics =
 			shuffle_extrinsics::<<Block as BlockT>::Extrinsic>(extrinsics, shuffling_seed);
 
-		let block_builder = BlockBuilder::new(
+		let parent_hash = self.client.info().best_hash;
+		let parent_number = self.client.info().best_number;
+
+		let mut block_builder = BlockBuilder::new(
 			&*self.runtime_api,
-			self.client.info().best_hash,
-			self.client.info().best_number,
+			parent_hash,
+			parent_number,
 			RecordProof::No,
 			Default::default(),
 			&*self.backend,
-			extrinsics,
 		)?;
+
+		let inherent_data = self.inherent_data(parent_hash, primary_hash).await?;
+
+		let mut final_extrinsics = block_builder.create_inherents(inherent_data)?;
+		final_extrinsics.extend(extrinsics);
+
+		block_builder.set_extrinsics(final_extrinsics);
```

Comment:
This feels like an awkward API, can't we call `.create_inherents()` and set all of the extrinsics in the correct order in `.build()` itself? Should be less error-prone API.

Reply:
Hmm, you mean `build(inherent_data, extrinsics)`?

Reply:
Yeah, something like that. I checked the code, there seems to be just this place and one test where `.build()` is called, so I'm not sure if we want to create inherents in that test automatically. I see that API in Substrate is similar to what to have with `.push()` method, just wondering if this is the best way to do it.